rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -309,4 +303,42 @@ void test_extractRedeemScriptFromInput_migration_multiple_inputs() {
             Assertions.assertEquals(expectedRedeemScript, redeemScript.get());
         }
     }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionIsSegwit_shouldThrowIllegalArgumentException() {
+        // arrange
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        TransactionWitness transactionWitness = new TransactionWitness(1);
+        transactionWitness.setPush(0, new byte[]{});
+        transaction.setWitness(0, transactionWitness);
+
+        // act & assert
+        IllegalArgumentException exception =
+            Assertions.assertThrows(IllegalArgumentException.class, () -> BitcoinUtils.removeSignaturesFromNonSegwitTransaction(transaction));
+        Assertions.assertEquals("Removing signatures from SegWit transactions is not allowed.", exception.getMessage());
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionIsLegacy_shouldRemoveSignaturesFromScriptSig() {
+        // arrange
+        Federation federation = new P2shErpFederationBuilder().build();
+        Script scriptSig = federation.getP2SHScript().createEmptyInputScript(null, federation.getRedeemScript());
+
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, scriptSig);
+        
+        BtcTransaction transactionWithoutSignatures = new BtcTransaction(btcMainnetParams, transaction.bitcoinSerialize());
+
+        for (int i = 0; i < federation.getNumberOfSignaturesRequired(); i++) {
+            BtcECKey key = new BtcECKey();
+            BitcoinTestUtils.addSignaturesToInput(transaction, 0, Collections.singletonList(key));
+        }
+
+        // act
+        BitcoinUtils.removeSignaturesFromNonSegwitTransaction(transaction);
```

Comment:
Since `addSignaturesToInput` is a test utility method and doesn't have test for itself, should we check that `transaction` really has signatures before trying to remove them?